### PR TITLE
J2ee10 - remove_parm soap_element_factory recipes

### PR DESCRIPTION
**Pull request by ranuradh**
## What's changed?
image

image
## Anything in particular you'd like reviewers to focus on?
@timtebeek @joanvr

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've added the license header to any new files through `./gradlew licenseFormat`
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/classpath/method_expression_test.jar
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: Can we move these files to `src/test/resource` instead? I don't think they are necessary at runtime seeing as we only have declarative recipes in yaml.

Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: sure will do that

Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: @timtebeek should  I create a folder src/test/resource and then add the files there? I only see "src/test/resources" just want to make sure

Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: Done..I wasn't sure where to put it next time will make sure its in this folder. Thanks

**Review by joanvr — APPROVED** [review]

**Review by joanvr — APPROVED** [review]
I'm seeing a failed test, missing a recipe called `org.openrewrite.java.migrate.jakarta.RemoveNewInstance`?

File: src/test/resources/META-INF/rewrite/classpath/method_expression_test.jar
Comment (joanvr) [inline src/test/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: why we have this file? I've checked the contents and it's just a Test1 class that it's not being used and a stub for MethodExpression. Why don't we add the actual jakarta.el-api jar or add the stub in the test itself?

File: src/test/java/org/openrewrite/java/migrate/jakarta/RemovedSOAPElementFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    Name n = envelope.createName("GetLastTradePrice", "WOMBAT", "http://www.abc.org/trader");
+                    SOAPElementFactory sfe = null;
+                    sfe.create(str1);
+                    sfe.create(str1, str2, str3);
+                    sfe.create(n);
+                    sfe.newInstance();
+                }
+                void doY() {
+                    String x = "test";
+                    x.toUpperCase();
```
Comment (joanvr) [inline src/test/java/org/openrewrite/java/migrate/jakarta/RemovedSOAPElementFactoryTest.java:62]: what's the purpose of this extra stuff in the test?

**Comment by joanvr** [comment]
After a bit more of diving into the recipe:
SOAPElementFactory.newInstance() method has no direct replacement but probably should be replaced by SOAPFactory.newInstance() when you are replacing the whole type? 
I'm not sure about this part of the recipe

**Comment by joanvr** [comment]
Pushed some changes:
- Improved format on tests
- Removed unnecessary bits on tests
- Removed multiple versions of jakarta.xml.soap-api
- Added jakarta.el-api
- Removed recipe that removes SOAPElementFactory.newInstance() -> Instead just let it be changed to SOAPFactory.newInstance()

File: src/test/java/org/openrewrite/java/migrate/jakarta/RemovedSOAPElementFactoryTest.java (hunk 1 above)
Comment (ranuradh) [reply src/test/java/org/openrewrite/java/migrate/jakarta/RemovedSOAPElementFactoryTest.java]: This extra stuff was just to ensure nothing happened to it and I have used it as a negative test..

File: src/test/resources/META-INF/rewrite/classpath/method_expression_test.jar
Comment (ranuradh) [reply src/test/resources/META-INF/rewrite/classpath/method_expression_test.jar:1]: yes we can use the jakarta.el-api.jar instead

**Comment by ranuradh** [comment]
I approve the slight test changes and adding the .jars. ,Regarding The static method **SOAPElementFactory.newInstance(**) is removed without replacement. See the [Jakarta EE 9.1 documentation](https://jakarta.ee/specifications/platform/9.1/apidocs/jakarta/xml/soap/soapelementfactory) for more information about the replacement for this removed class. We need to remove it not replace it that is why I had created a new class.

**Comment by joanvr** [comment]
Merged! Thank you very much for your time and contributions @ranuradh :)

**Pull request merged**